pkg/resource/notebook_instance/hooks.go

```diff
@@ -0,0 +1,49 @@
+package notebook_instance
+
+import (
+	"context"
+	"errors"
+
+	ackrequeue "github.com/aws-controllers-k8s/runtime/pkg/requeue"
+	svccommon "github.com/aws-controllers-k8s/sagemaker-controller/pkg/common"
+	svcsdk "github.com/aws/aws-sdk-go/service/sagemaker"
+)
+
+var (
+	modifyingStatuses = []string{
```

Comment:
minor nit: modifyingStatuses is a slightly confusing name though it becomes obvious when you look at the actual list - so ok as is for now.